src/api/BundlePush/BundlePusher.ts

```diff
@@ -527,4 +609,87 @@ export class BundlePusher {
       await this.runSingleNpmUninstall(sshSession, remoteDirectory);
     }
   }
+
+  private async outputNodejsDiagnostics(cicsSession: AbstractSession, dfhdployOutput: string) {
+    // Did the Bundle contents include a NODEJSAPP?
+    if (dfhdployOutput.indexOf("http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP") > -1) {
+      let diagnosticsIssued = false;
+      try {
+        if (cicsSession !== undefined) {
+          // Attempt to gather additional Node.js specific information from CICS
+          this.updateStatus("Gathering Node.js diagnostics");
+          diagnosticsIssued = await this.gatherNodejsDiagnosticsFromCics(cicsSession);
+        }
+      }
+      catch (diagnosticsError) {
+        // Something went wrong generating diagnostic info. Don't trouble the user
+        // with what might be an exotic error message (e.g. hex dump of an entire HTML page),
+        // just log the failure.
+        if (this.params.arguments.silent === undefined) {
+          const logger = Logger.getAppLogger();
+          logger.debug(diagnosticsError.message);
+        }
+      }
+
+      // If we don't have a cics profile or if we do but the diagnostics collection failed, issue a message.
+      if (diagnosticsIssued === false) {
+        const msg = "For further information on the state of your NODEJSAPP resources, consider running the following command:\n\n" +
+              "zowe cics get resource CICSNodejsapp --region-name " + this.params.arguments.scope +
+              " --criteria \"BUNDLE=" + this.params.arguments.name + "\" --cics-plex " + this.params.arguments.cicsplex + "\n";
+        this.issueMessage(msg);
+      }
+    }
+  }
+
+  private async gatherNodejsDiagnosticsFromCics(cicsSession: AbstractSession): Promise<boolean> {
+    // Issue a CMCI get to the target CICSplex
+    try {
+      const data: IResourceParms = { name: "CICSNodejsapp",
+                                     criteria: "BUNDLE=" + this.params.arguments.name,
+                                     regionName: this.params.arguments.scope,
+                                     cicsPlex: this.params.arguments.cicsplex };
+      const cmciResponse = await getResource(cicsSession, data);
+      const outputRecord = cmciResponse.response.records.cicsnodejsapp;
```

Comment:
Should we guard against `cmciReponse.response` and `cmciReponse.records` being `undefined` here too?